[IMP] base: add thread_time to profiler

kmagusiak opened the pull request:
Adding cpu_duration to ir.profile that contains the duration based on `time.thread_time`. This number indicates how much CPU time was spent by the request excluding other processes and sleep time. It can be used to distinguish at a single glance between time spent in the python process and time spent elsewhere (process context switching, database, etc.).

Duration fields are now stored with a precision down to the millisecond. They used to be stored as float and displayed only 2 digits.



---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

Julien00859 reviewed (approved):
@Xavier-Do

Xavier-Do reviewed (approved):
LGTM but the help message could be maybe explain that it represents the python execution time and doesn't count sql time.

The current message "CPU clock (not including other processes)" could suggest that it is the real execution time of the request, not counting parallel request and it is not the case.

It could also be interesting to add this in the view form as optional=false

kmagusiak commented:
> LGTM but the help message could be maybe explain that it represents the python execution time and doesn't count sql time.
> 
> The current message "CPU clock (not including other processes)" could suggest that it is the real execution time of the request, not counting parallel request and it is not the case.
> 
> It could also be interesting to add this in the view list as optional=false

I updated the help message and added optional="hide" (not sur false makes sense).

The counted time is thread-based `real_cpu_time = time.thread_time.__call__` so it really measures one request, because when starting to process a request, the thread is only doing that.

Xavier-Do commented:
@robodoo r+
